### DISCUSSION TITLE
Как создать токен, если используется двухфакторная аутентификация?

### DISCUSSION THREAD

**Discussion by Yanpas** [MarshalX/yandex-music-api#385] [Q&A] [answered]
Я так понимаю единственный вариант - это токен. Создание отладочного токена для моего же приложения раньше помогало, а сейчас этот токен уже не работает:
 ```
Found unknown fields received from API! Please copy warn message and send to https://bit.ly/3dsFxyH (github issue), thank you!
Type: <class 'yandex_music.account.status.Status'>; kwargs: {'advertisement': 'Оформите постоянную подписку – первый месяц бесплатно!', 'bar_below': {'alert_id': 'xxx', 'text': 'тест Подписка скоро закончится. А вместе с ней закончатся музыка и подкасты без ограничений. Не надо так.', 'bg_color': '#FF5038', 'text_color': '#FFFFFF', 'alert_type': 'Churn_SubscriptionEnd_Music', 'button': {'text': 'Оформить подписку', 'bg_color': '#FFFFFF', 'text_color': '#191919', 'uri': 'https://plus.yandex.ru/?source=music_web_churn_subscriptionend'}, 'close_button': False}}
```

**Comment by MarshalX** [comment, marked as answer]
При 2ФА вместе со своим логином можно использовать одноразовые пароли, которые используются вместо обычных. Они генерируются в приложении от Яндекс.

Предупреждения сообщают о новых полях в API – это не ошибка. Работе не мешает. Можно отключить, а на `development` ветке выключено по умолчанию.

**Reply by Maxbudkov** [reply]
Спасибо за вашу огромную работу. Использовать одноразовые ключи при 2ФА можно, но что насчет постоянного использования? Если я пытаюсь сделать несколько запросов к апи яндекса? Я только начал изучать ваш проект и многого ещё не понимаю, не судите строго. И что делать в случае ошибки 403 - Capcha Required?

**Reply by MarshalX** [reply]
Для постоянного использования нужно один раз авторизоваться через логин и пароль, запомнить токен авторизации, а при следующем использовании использовать токен. В случае необходимости пройти капчу надо её обработать и ввести. В главном ридми есть пример обработки капчи…

**Reply by Maxbudkov** [reply]
Большое спасибо за уточнение (про капчу действительно не дочитал)
